mzLib/Omics/ISpectralMatch.cs

```diff
@@ -0,0 +1,97 @@
+﻿
+
+namespace Omics
+{
+    /// <summary>
+    /// Represents a single spectrum-to-sequence identification (PSM) produced by a reader or search engine.
+    /// Implementations provide identifying sequence strings, a numeric confidence score, the originating
+    /// file identifier, and access to the biopolymer objects (peptides/proteoforms) that were assigned to
+    /// the spectrum. Implementations must provide a consistent comparison implementation so matches can
+    /// be ordered when needed.
+    /// </summary>
+    public interface ISpectralMatch : IComparable<ISpectralMatch>
+    {
+        /// <summary>
+        /// The (absolute or relative) file path or file identifier for the spectra file that produced this match.
+        /// Implementations may normalize this value (for example to an empty string) but should document whether
+        /// comparisons treat path differences (case or separators) specially.
+        /// </summary>
+        string FullFilePath { get; }
+
+        /// <summary>
+        /// The full modified sequence string in the reader/search format (includes modification annotations).
+        /// This is the canonical representation produced by the parser/search and should be used when
+        /// presenting or comparing modified sequences.
+        /// </summary>
+        string FullSequence { get; }
+
+        /// <summary>
+        /// The base (unmodified) sequence. Use this for sequence-level grouping and comparisons when
+        /// modification details are not relevant.
+        /// </summary>
+        string BaseSequence { get; }
+
+        /// <summary>
+        /// The scan number associated with this identification.
+        /// Convention is one-based indexing; implementers should document the indexing they use.
+        /// </summary>
+        int ScanNumber { get; }
```

Comment:
I think this should explicitly be defined as "OneBasedScanNumber". Implementers should not get to choose the type of indexing, it should be standardized